### PR TITLE
Add template CFF

### LINKED ISSUE #508: Feature request: Adopt Citation File Format (CITATION.cff file)
There is a nice standard to express citation metadata information in a structured way for e.g. software or data publications: https://citation-file-format.github.io/

This format also has support by Github and is supported by Github-to-Zenodo pipelines: 

* https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files
* https://the-turing-way.netlify.app/communication/citable/citable-cff.html
* https://docs.software-metadata.pub/en/latest/tutorials/automated-publication-with-ci.html

There are also already visual tools to help filling the file with information: https://citation-file-format.github.io/cff-initializer-javascript/#/

Also, there are tools to check its validity (e.g. usable as part of the CI): https://github.com/citation-file-format/cffconvert/tree/main/cffconvert

It would be great if the Carpentries would adopt this best-practice standard by using `CITATION.cff` as the way to state the citation information, instead of linking to a (possibly non-existing) plain-text `CITATION` file.

Ideally, all courses would have a valid `CITATION.cff` file and this would be part of the standard template from which a lesson is initialized.

The minimal suggestion would be to at least allow relinking the "Cite" link at the bottom to point to a CITATION.cff file.

### PR DESCRIPTION
**Pull request by tobyhodges**
Closes #508 

Adds a template `CITATION.cff` file to be included in new lessons initialised with `create_lesson()`. I had a go at writing tests, and would love feedback on what I missed and how I could have done it better.

### CODE REVIEW

File: inst/templates/citation-template.txt
[the remark this answers is not in the document]
Comment (tobyhodges) [reply inst/templates/citation-template.txt:1]: Thanks @sdruskat I had considered adding this field, and other fields I would recommend such as `identifiers`, but in the end opted for a "minimum viable lesson CFF". I am trying to supplement this template content with documentation elsewhere (e.g. in the Collaborative Lesson Development Training curriculum and the Workbench docs). 

TBH we are also still figuring out exactly what guidance we want to give the community about the information they put into their CFFs: creating them is the easy part, but updating them as the lesson develops and matures is more tricky...

I'm afraid the kind of automated filling of information you mentioned is not possible right now, but would indeed be a cool development for the future.

[the remark this answers is not in the document]
Comment (ocaisa) [reply inst/templates/citation-template.txt:1]: One thing I've asked is how do we cross-reference to another CFF (adding tons of authors would make this cumbersome). I got a response for this in https://github.com/citation-file-format/citation-file-format/issues/538#issuecomment-2051488372 which I think may be useful (CI is one option, reference to the CFF file directly is another)

[the remark this answers is not in the document]
Comment (tobyhodges) [reply inst/templates/citation-template.txt:1]: Thanks @ocaisa. I guess I need to open a new issue somewhere to keep track of what advice we should give people about CFF, which would include your suggestion. I think docs.carpentries.org is the appropriate place.

**Comment by froggleston** [comment]
Looks great! 🚀

**Pull request merged**